foodsaving/stores/serializers.py

```diff
@@ -229,3 +232,40 @@ def validate_weeks_in_advance(self, w):
         if w < 1:
             raise serializers.ValidationError(_('Set at least one week in advance'))
         return w
+
+
+class FeedbackSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = FeedbackModel
+        fields = ['id', 'weight', 'comment', 'about', 'given_by']
+        # read_only_fields = ('given_by',)
+
+        """
+        Tilmann suggested to make 'given_by' read_only - but the following code produces errors
+        extra_kwargs = {
```

Comment:
I would be interested to see what errors you get. Probably the database complains that the `given_by` field is missing a value, as it is non-nullable.

To solve this issue, you need to provide then `given_by` field when creating the entry. Override this `create()` method for that.

```
    def create(self, validated_data):
        validated_data['given_by'] = self.context['request'].user
        return super().create(validated_data)
```

I just noticed that you discovered the new `read_only_fields` [shortcut](http://www.django-rest-framework.org/api-guide/serializers/#specifying-read-only-fields). Great, please use that instead of `extra_kwargs`!

Reply:
Yes, we are getting errors also for `read_only_fields`. We will try it then with `create()` to avoid errors.